services/housekeeper/housekeeper.go

```diff
@@ -171,6 +179,49 @@ func (hk *Housekeeper) updateProposerDuties(headSlot uint64) {
 	hk.UpdateProposerDutiesWithoutChecks(headSlot)
 }
 
+func (hk *Housekeeper) monitorMevCommitBuilderRegistrations() {
+	newBuilderRegistered, err := hk.mevCommitClient.ListenForActiveBuildersEvents()
+	if err != nil {
+		hk.log.WithError(err).Error("failed to subscribe to MEV-Commit builder registered events")
+		return
+	}
+
+	for {
+		builder := <-newBuilderRegistered
+		hk.log.WithField("builder", builder).Info("new builder registered")
+		err := hk.redis.SetMevCommitBlockBuilder(builder)
+		if err != nil {
+			hk.log.WithError(err).Error("failed to set MEV-Commit builder registration")
+		}
+	}
+}
+
+func (hk *Housekeeper) cleanupMevCommitBuilderRegistrations() {
```

Comment:
Its fine to do this once in the start. But we need to monitor `FundsSlashed` event. Based on this event we need to Delete the registration. Currently we only check when we initially set it, however we might see providers getting their stakes slashed while this is running. So we need to handle that.